### ISSUE TITLE
[BUG] Missing KQueueEventLoopGroup from azure-core-http-netty

### ISSUE THREAD

**Issue by omarsmak** [Azure/azure-sdk-for-java#12170]
Hello, 

Sometimes, I hit this error:
```
java.lang.ClassCastException: io.netty.channel.kqueue.KQueueEventLoopGroup cannot be cast to io.netty.channel.EventLoopGroup
	at reactor.netty.resources.DefaultLoopResources.cacheNativeServerLoops(DefaultLoopResources.java:249)
	at reactor.netty.resources.DefaultLoopResources.cacheNativeServerLoops(DefaultLoopResources.java:259)
	at reactor.netty.resources.DefaultLoopResources.cacheNativeClientLoops(DefaultLoopResources.java:267)
	at reactor.netty.resources.DefaultLoopResources.onClient(DefaultLoopResources.java:199)
	at reactor.netty.tcp.TcpResources.onClient(TcpResources.java:218)
	at reactor.netty.http.client.HttpClientConnect$HttpTcpClient.connect(HttpClientConnect.java:135)
	at reactor.netty.tcp.TcpClientOperator.connect(TcpClientOperator.java:43)
	at reactor.netty.tcp.TcpClientOperator.connect(TcpClientOperator.java:43)
	at reactor.netty.tcp.TcpClientOperator.connect(TcpClientOperator.java:43)
	at reactor.netty.tcp.TcpClientOperator.connect(TcpClientOperator.java:43)
	at reactor.netty.tcp.TcpClientOperator.connect(TcpClientOperator.java:43)
	at reactor.netty.tcp.TcpClient.connect(TcpClient.java:189)
	at reactor.netty.http.client.HttpClientFinalizer.connect(HttpClientFinalizer.java:68)
	at reactor.netty.http.client.HttpClientFinalizer.responseConnection(HttpClientFinalizer.java:85)
	at com.azure.core.http.netty.NettyAsyncHttpClient.send(NettyAsyncHttpClient.java:95)
	at com.azure.core.http.HttpPipelineNextPolicy.process(HttpPipelineNextPolicy.java:44)
	at com.azure.storage.common.policy.ScrubEtagPolicy.process(ScrubEtagPolicy.java:35)
	at com.azure.core.http.HttpPipelineNextPolicy.process(HttpPipelineNextPolicy.java:46)
```

No idea why it is happening, after some debugging I did, I realized `KQueueEventLoopGroup` was available in netty package `netty-transport-native-kqueue` but this package wasn't add it as dependency in the pom file of [azure-core-http-netty](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-http-netty/pom.xml#L59) which resulted of the above error.
To solve this issue from my end, I had to explicitly add the dependency of  `netty-transport-native-kqueue` to my pom file:
```
 <dependency>
            <groupId>io.netty</groupId>
            <artifactId>netty-transport-native-kqueue</artifactId>
            <version>4.1.45.Final</version>
        </dependency>
```

Is this something is known? 
Thank you

**Comment by alzimmermsft** [comment]
Thank you for creating this issue.

One solution for this issue would be adding another classified dependency in `azure-core-http-netty` for the KQueue transport plugin as was already done for Linux's EPoll.

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-http-netty/pom.xml#L98

**Comment by alzimmermsft** [comment]
Fixed in #12503

**Comment by omarsmak** [comment]
awesome! thanks @alzimmermsft !
